Engineering/Frontend/WAI-ARIA.md

```diff
@@ -0,0 +1,70 @@
+---
+tags: engineering/frontend, wai-aria, accessibility
+author: Nguyen Dinh Nam
+date: 2022-09-30
+---
+
+Since modern websites and applications contain huge amounts of dynamic content and complex components with no semantics to describe what they mean, users with **no mouse** equiped or those dependent on technologies such as **[Screen Reader](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn/related?hl=en)** might not be able to interact with those components.
+
+Web Accessibility Initiative’s Accessible Rich Internet Applications or simply **WAI-ARIA** is a technology that can help with such problems.
+
+## What is WAI-ARIA?
+Before semantic elements like `<nav>` or `<footer>` were introduced that define specific features of a web page, some developers would rely on JavaScript libraries that generate a bunch of **nested `<div>`s**, then styled them with CSS and controlled with JavaScript. 
+
+Even though they work fine, screen readers cannot make any sense of what those components are, no semantics provided so that they can assist their users. WAI-ARIA has **roles**, **states** and **properties** to solve this problem, giving an opportunity to add attributes to content and components that make them **meaningful** and enhance **accessibility**.
+
+### Roles:
+[WAI-ARIA roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles) define **what that HTML is or does**, add the required additional information in cases the markup does not provide the required role. If we write `<a role="button">`, the screen reader will recognize it as a button.
```

Comment:
You might want to expand more on what a screen reader is (e.g. more ref link). Many readers may not know what it is yet.

Reply:
Keep in mind that we want to introduce **screen reader** as a concept, not just any extension.